packages/shell-chrome/src/backend.js

```diff
@@ -86,7 +86,8 @@ function handleMessages(e) {
                         data[attributeSequence] = attributeValue
                     }
 
-                    component.__x.$el.setAttribute('x-data', JSON.stringify(data))
+                    component.__x.$data = data
+                    component.__x.updateElements(component)
```

Comment:
This is possibly a big change so I want to point it out.

Also, on the Inline devtools I try to only update the specific property which I think is much faster, so maybe something to tackle here on another day
https://github.com/KevinBatdorf/alpine-inline-devtools/blob/master/src/Viewer.js#L69

Reply:
I think this will be fine, was not a fan of overwriting `x-data`

Reply:
On second thought, would it be possible to make this a separate PR/issue?

Would like to release this `HTMLElement` fix without changing the way we update user's component state.

Reply:
I like the idea of only updating a single field, could be something like

```js
if (attributeSequence.includes('.')) {
  set(component.__x.$data, attributeSequence, attributeValue)
  component.__x.updateElements(component) // force re-render on Array/Object update?
} else {
  component.__x.$data[attributeSequence] = attributeValue
}
```

but let's do that as a separate performance thing

Reply:
ok I'll remove this and we can have another look in https://github.com/alpine-collective/alpinejs-devtools/issues/97